features/cellular/framework/API/CellularContext.h

```diff
@@ -121,6 +121,8 @@ class CellularContext : public CellularInterface {
 public: // from NetworkInterface
     virtual nsapi_error_t set_blocking(bool blocking) = 0;
     virtual NetworkStack *get_stack() = 0;
+    virtual nsapi_error_t get_ip_address(SocketAddress *address) = 0;
+    MBED_DEPRECATED_SINCE("mbed-os-5.15", "String-based APIs are deprecated")
```

Comment:
I think you are deprecating the wrong thing, that should go before the call afaik.